### PR TITLE
[1910] Ruby 2.7 => 3.1.2

### PR DESCRIPTION
**Pull request by d-a-v-e**
https://eaflood.atlassian.net/browse/RUBY-1910

### CODE REVIEW

File: app/models/enrollment_export.rb [hunk 1]
```diff
@@ -14,28 +14,18 @@ class EnrollmentExport < ActiveRecord::Base
...
-
-    case date_field_scope
-      when "submitted_at"
-        find_klazz.reportable_by_submitted_at(from_date, to_date)
-      when "decision_at"
-        find_klazz.reportable_by_decision_at(from_date, to_date)
-      else
-        find_klazz.reportable_by_submitted_at(from_date, to_date)
-    end
+    FloodRiskEngine::EnrollmentExemption.send("reportable_by_#{date_field_scope}", from_date, to_date)
```
Comment (PaulDoyle-DEFRA) [inline app/models/enrollment_export.rb:28]: nice

File: spec/models/enrollment_export_spec.rb [hunk 2]
```diff
@@ -48,29 +48,28 @@
 
     context "on_or_before" do
       let(:enrollment_export) { build(:enrollment_export) }
-
-      subject { enrollment_export.valid? }
```
Comment (PaulDoyle-DEFRA) [inline spec/models/enrollment_export_spec.rb:52]: I can't see where the subject is being defined/called in the updated code

File: app/models/enrollment_export.rb [hunk 3]
```diff
@@ -14,28 +14,18 @@ class EnrollmentExport < ActiveRecord::Base
     decision_at: 1
   }
 
-  validates :from_date, presence: true
-  validates_date :from_date, on_or_before: :today, allow_blank: true
```
[the remark this answers is not in the document]
Comment (d-a-v-e) [reply app/models/enrollment_export.rb:18]: These were validators from the removed `validates_timeliness` gem. I've replicated the logic in custom validations (from line 86).

[the remark this answers is not in the document]
Comment (d-a-v-e) [reply app/models/enrollment_export.rb:18]: the allow blank seems to contradicts the `presence: true` so I've left it off

File: spec/models/enrollment_export_spec.rb (hunk 2 above)
Comment (d-a-v-e) [reply spec/models/enrollment_export_spec.rb:52]: this was just being used to trigger the validation (which just returns a true/false) and not being read.

**Review by PaulDoyle-DEFRA — APPROVED** [review]

**Pull request merged**